src/Npgsql/NpgsqlSchema.cs

```diff
@@ -563,6 +566,247 @@ static DataTable GetDataSourceInformation(NpgsqlConnection conn)
+                    row["DataType"] = baseMapping.ClrTypes[0].Name;
+                if (baseMapping.NpgsqlDbType.HasValue)
+                    row["ProviderDbType"] = (int)baseMapping.NpgsqlDbType.Value;
+            }
+
+            return table;
+        }
+
+        /// <summary>
+        /// Populates some generic type information that is common for base types, arrays, enums, etc. Some will
+        /// be overridden later.
+        /// </summary>
+        static void PopulateDefaultDataTypeInfo(DataRow row, PostgresType type)
+        {
+            row["TypeName"] = type.DisplayName;
+            // Skipping ColumnSize at least for now, not very meaningful
+            row["CreateFormat"] = type.DisplayName.ToUpperInvariant();
+            row["CreateParameters"] = "";
+            row["IsAutoIncrementable"] = false;
+            // We populate the DataType above from mapping.ClrTypes, which means we take the .NET type from
+            // which we *infer* the PostgreSQL type. Since only a single PostgreSQL type gets inferred from a given
+            // .NET type, we never have the same DataType in more than one row - so the mapping is always the
+            // best match. See the hardcoding override  below for some exceptions.
+            row["IsBestMatch"] = true;
+            row["IsCaseSensitive"] = true;
+            row["IsConcurrencyType"] = false;
+            row["IsFixedLength"] = false;
+            row["IsFixedPrecisionAndScale"] = false;
+            row["IsLiteralSupported"] = false;  // See hardcoding override below
+            row["IsLong"] = false;
+            row["IsNullable"] = true;
+            row["IsSearchable"] = true;
+            row["IsSearchableWithLike"] = false;
+            row["IsUnsigned"] = DBNull.Value; // See hardcoding override below
+            // LiteralPrefix: no literal for now except for strings, see hardcoding override below
+            // LiteralSuffix: no literal for now except for strings, see hardcoding override below
+            row["MaximumScale"] = DBNull.Value;
+            row["MinimumScale"] = DBNull.Value;
+            // NativeDataType is unset
+            row["OID"] = type.OID;
+        }
+
+        /// <summary>
+        /// Sets some custom, hardcoded info on a DataType row that cannot be loaded/inferred from PostgreSQL
+        /// </summary>
+        static void PopulateHardcodedDataTypeInfo(DataRow row, PostgresType type)
+        {
+            if (type.Name == "text" || type.Name == "varchar" || type.Name == "char")
+            {
+                row["CreateFormat"] += "({0})";
+                row["CreateParameters"] = "size";
+                row["IsSearchableWithLike"] = true;
+                row["IsLiteralSupported"] = true;
+                row["LiteralPrefix"] = "'";
+                row["LiteralSuffix"] = "'";
+
+                if (type.Name == "varchar" || type.Name == "char")
+                {
+                    row["DataType"] = "String";
```

Comment:
You could use `typeof(string).Name` or `nameof(System.String)` here to follow the pattern from above where you assign via `.ClrTypes[0].Name`. Not that it changes anything semantically, but it might look more consistent down the road.

Reply:
Not sure I see the value of typeof/nameof in this context... this is a hardcoded setting that will never, ever change...